platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/camera/CameraPosition.java

```diff
@@ -27,7 +29,9 @@ public CameraPosition createFromParcel(Parcel in) {
         LatLng target = in.readParcelable(LatLng.class.getClassLoader());
         double tilt = in.readDouble();
         double zoom = in.readDouble();
-        return new CameraPosition(target, zoom, tilt, bearing);
+        double[] padding = new double[4];
```

Comment:
nit hardcoding array size, it's better to store size of array upfront and read it out with `in.readDouble()`. We aren't enforcing an array size of 4 so a user could potentially pass a different size array.

Reply:
> We aren't enforcing an array size of 4 so a user could potentially pass a different size array.

We are asserting that in JNI and `@Size` should improve that as well. In this case, do you think that storing array size separately is still necessary?

Reply:
afaik asserts are removed with release builds and `@Size` gives developers a hint but doesn't enforce anything.

Reply:
Adde `@Size` to the `CameraPosition#Builder`. If not obeyed, `ArrayIndexOutOfBoundsException` is going to get thrown at some point during the conversion.